legate/core/store.py

```diff
@@ -158,6 +159,8 @@ def attach_external_allocation(
         assert self.physical_region_refs == 0
         # Record the attached memory ranges, and confirm no overlaps with
         # previously encountered ranges.
+        if self.detach_future and not self.detach_future.is_ready():
+            self.detach_future.wait()
```

Comment:
I don't think this is the right place to do this check. it should be done when the region field is recycled. and the recycling logic should be tweaked such that it favors free region fields with no pending detach over those with one.

Reply:
@magnatelee : I believe I have added requested logic. Do you think I should still keep the check here? Can there be the case when RegionField is re-attached without going through the FieldManager ?

Reply:
This if-statement should be changed to an assertion or an exception, as the store at this line shouldn't have any outstanding detachment. Also, re-attaching is illegal. Look at the check at line 160.